packages/default/test/renderers/text.field.test.tsx

```diff
@@ -343,3 +343,239 @@ test('enabled by default', t => {
   const input = findRenderedDOMElementWithTag(tree, 'input') as HTMLInputElement;
   t.false(input.disabled);
 });
+
+test('use maxLength for attributes size and maxlength', t => {
+    const schema = {
+        'type': 'object',
+        'properties': {
+            'name': {
+                'type': 'string',
+                'maxLength': 5
+            }
+        }
+    };
+    const uischema = {
+      'type': 'Control',
+      'scope': {
+        '$ref': '#/properties/name'
+      },
+      'options': {
+        'trim': true,
+        'restrict': true
+      }
+    };
+    const store = initJsonFormsStore(t.context.data, schema, uischema);
+    const tree = renderIntoDocument(
+        <Provider store={store}>
+            <TextField schema={schema} uischema={uischema}/>
+        </Provider>
+    );
+    const input = findRenderedDOMElementWithTag(tree, 'input') as HTMLInputElement;
+    t.is(input.maxLength, 5);
+    t.is(input.size, 5);
+});
+
+test('use maxLength for attribute size only', t => {
+  const schema = {
+    'type': 'object',
+    'properties': {
+      'name': {
+        'type': 'string',
+        'maxLength': 5
+      }
+    }
+  };
+  const uischema = {
+    'type': 'Control',
+    'scope': {
+      '$ref': '#/properties/name'
+    },
+    'options': {
+      'trim': true
+    }
+  };
+  const store = initJsonFormsStore(t.context.data, schema, uischema);
+  const tree = renderIntoDocument(
+    <Provider store={store}>
+      <TextField schema={schema} uischema={uischema}/>
+    </Provider>
+  );
+  const input = findRenderedDOMElementWithTag(tree, 'input') as HTMLInputElement;
+  t.is(input.maxLength, 524288);
```

Comment:
While we're at it I think we should extract 524288 to a constant and give it a proper name.

Reply:
yes would make sense